core/services/vrf/delegate_test.go

```diff
@@ -88,91 +108,226 @@ func generateCallbackReturnValues(t *testing.T) []byte {
+	assert.Equal(t, 1, len(listener.reqs))
+	// It should be confirmed at 10+6*(2^2)
+	assert.Equal(t, uint64(34), listener.reqs[0].confirmedAtBlock)
+}
+
+func TestDelegate_ValidLog(t *testing.T) {
+	vuni, listener, jb := setup(t)
+	txHash := utils.NewHash()
+	reqID1 := utils.NewHash()
+	reqID2 := utils.NewHash()
+	pk, err := secp256k1.NewPublicKeyFromHex(vuni.vrfkey.String())
+	require.NoError(t, err)
+	added := make(chan struct{})
+	listener.reqAdded = func() {
+		added <- struct{}{}
+	}
+	preSeed := common.BigToHash(big.NewInt(42)).Bytes()
+	bh := utils.NewHash()
+	var tt = []struct {
+		reqID [32]byte
+		log   types.Log
+	}{
 		{
-			// Data has all the NON-indexed parameters
-			Data: bytes.Join([][]byte{
-				pk.MustHash().Bytes(),                    // key hash
-				common.BigToHash(big.NewInt(42)).Bytes(), // seed
-				cltest.NewHash().Bytes(),                 // sender
-				cltest.NewHash().Bytes(),                 // fee
-				reqID.Bytes()},                           // requestID
-				[]byte{}),
-			// JobID is indexed, thats why it lives in the Topics.
-			Topics: []common.Hash{{}, jb.ExternalIDEncodeStringToTopic()}, // jobID STRING
-			TxHash: txHash,
+			reqID: reqID1,
+			log: types.Log{
+				// Data has all the NON-indexed parameters
+				Data: bytes.Join([][]byte{
+					pk.MustHash().Bytes(),                    // key hash
+					common.BigToHash(big.NewInt(42)).Bytes(), // seed
+					utils.NewHash().Bytes(),                  // sender
+					utils.NewHash().Bytes(),                  // fee
+					reqID1.Bytes()},                          // requestID
+					[]byte{}),
+				// JobID is indexed, thats why it lives in the Topics.
+				Topics:      []common.Hash{{}, jb.ExternalIDEncodeStringToTopic()}, // jobID STRING
+				TxHash:      txHash,
+				BlockNumber: 10,
+				BlockHash:   bh,
+			},
 		},
 		{
-			Data: bytes.Join([][]byte{
-				pk.MustHash().Bytes(),                    // key hash
-				common.BigToHash(big.NewInt(42)).Bytes(), // seed
-				cltest.NewHash().Bytes(),                 // sender
-				cltest.NewHash().Bytes(),                 // fee
-				reqID.Bytes()},                           // requestID
-				[]byte{}),
-			Topics: []common.Hash{{}, jb.ExternalIDEncodeBytesToTopic()}, // jobID BYTES
-			TxHash: txHash,
+
+			reqID: reqID2,
+			log: types.Log{
+				Data: bytes.Join([][]byte{
+					pk.MustHash().Bytes(),                    // key hash
+					common.BigToHash(big.NewInt(42)).Bytes(), // seed
+					utils.NewHash().Bytes(),                  // sender
+					utils.NewHash().Bytes(),                  // fee
+					reqID2.Bytes()},                          // requestID
+					[]byte{}),
+				Topics:      []common.Hash{{}, jb.ExternalIDEncodeBytesToTopic()}, // jobID BYTES
+				TxHash:      txHash,
+				BlockNumber: 10,
+				BlockHash:   bh,
+			},
 		},
 	}
 
 	consumed := make(chan struct{})
-	for i, l := range logs {
+	for i, tc := range tt {
+		tc := tc
```

Comment:
might be more readable to have some longer names than tt and tc

Reply:
meh in the go docs they use tt := range tests, I like tc := range tt (tc = test case)

Reply:
ah ok yeah makes sense